src/tests/functional_tests.rs

```diff
@@ -83,6 +83,22 @@ fn wait_for_tx(txid: Txid) {
 	}
 }
 
+fn wait_for_outpoint_spend(outpoint: OutPoint) {
+	let tx = get_electrsd().client.transaction_get(&outpoint.txid).unwrap();
+	let txout_script = tx.output.get(outpoint.vout as usize).unwrap().clone().script_pubkey;
+	let mut is_spent = !get_electrsd().client.script_get_history(&txout_script).unwrap().is_empty();
+	loop {
+		if is_spent {
+			break;
+		}
+		is_spent = exponential_backoff_poll(|| {
+			get_electrsd().trigger().unwrap();
+			get_electrsd().client.ping().unwrap();
+			Some(!get_electrsd().client.script_get_history(&txout_script).unwrap().is_empty())
+		});
```

Comment:
Will we ever give up? Shouldn't have the test hang if there is a problem either, I suppose.